shell/common/engine.cc

```diff
@@ -127,18 +131,80 @@ void Engine::OnViewportMetricsChanged(sky::ViewportMetricsPtr metrics) {
     runtime_->SetViewportMetrics(viewport_metrics_);
 }
 
-void Engine::OnLocaleChanged(const mojo::String& language_code,
-                             const mojo::String& country_code) {
-  language_code_ = language_code;
-  country_code_ = country_code;
-  if (runtime_)
-    runtime_->SetLocale(language_code_, country_code_);
+void Engine::DispatchPlatformMessage(
+    ftl::RefPtr<blink::PlatformMessage> message) {
+  if (message->channel() == kLifecycleChannel) {
+    if (HandleLifecyclePlatformMessage(message.get()))
+      return;
+  } else if (message->channel() == kLocalizationChannel) {
+    if (HandleNavigationPlatformMessage(std::move(message)))
+      return;
+  }
+
+  if (runtime_) {
+    runtime_->DispatchPlatformMessage(std::move(message));
+    return;
+  }
+
+  // If there's no runtime_, we need to buffer some navigation messages.
+  if (message->channel() == kNavigationChannel)
+    HandleNavigationPlatformMessage(std::move(message));
```

Comment:
I left the `dart:ui` interface for local unchanged whereas I removed the callbacks on `Window` for navigation.  The fact that we use platform messages to implement local is an implementation detail in this approach.

I could change that and expose the underlying platform messages for locale if you prefer.  The reason I picked this approach is because we have a nice `Locale` object that would be lost in the translation.